setup.py

```diff
@@ -1,7 +1,16 @@
 from setuptools import setup
+from pip._internal.req import parse_requirements
 
 VERSION = "0.0.11"
 
+requirements_py3 = [str(r.req) for r in parse_requirements('requirements.txt', session=False)]
+requirements_py2 = [str(r.req) for r in parse_requirements('requirements-py2.txt', session=False)]
+install_requires = [
+    '%s; python_version>="3"' % req for req in requirements_py3
+] + [
+    '%s; python_version<"3"' % req for req in requirements_py2
+]
+
```

Comment:
How does this work and what is it for?

Reply:
We have different requirements for python 2 & python 3

Reply:
I understand. I'm asking about the format of install requires :) Can you send any link where I can read about it?

Also, I'm a bit concerned that we import something from `pip._internal`. 
If we want to package for different python versions and have different requirements for each, what do you think about making it this way:
```
import sys

if sys.version_info.major == 2:
  # run setup for python 2
  setup(...,install_requires=requirements_py2)
else:
  # run setup for python 3
  setup(...,install_requires=requirements_py3)
```
?

Reply:
Sure, check out `setuptools` documentation: https://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-platform-specific-dependencies

> Also, I'm a bit concerned that we import something from pip._internal.
I've checked it in both python2.7 & python3.6

Regarding the suggestion, I think it makes sense once we'll have multiple conditionals on python version (e.g. not only `install_requires` but also `description` or another field).

Reply:
Gotcha.

Regarding `pip._internal`: well, if it works now, it doesn't mean that it'll not break in the future. Internal parts are not for public, it's not an intended use.

It seems that you only need `parse_requirements` from there, let's use https://setuptools.readthedocs.io/en/latest/pkg_resources.html#requirements-parsing then?

Reply:
Despite the similar name, these methods are quite different. In particular, the one from `pip` supports referencing (as we have in `requirements-py2.txt`) & extra indexes (`--extra-index-url`).
I agree that using internal components is generally a bad practice, but the worst thing we can get here is broken package deployment, which we can easily fix. So  I'm leaning towards keeping it as it is especially since we will get rid of it together once we migrated to `pipenv`. Wdyt?

Reply:
What's the problem of using conditional `setup(..)` based on `sys.version_info` then? It'll work & we don't use internal modules of pip.

Reply:
Discussed in slack, @persiyanov shared an awesome comment on difference between requirements file and install_requires in setup.py: https://stackoverflow.com/a/33685899